README.md

```diff
@@ -87,3 +87,129 @@ An ESX host can be attached and managed indirectly via vSphere API:
+static = High performance
+dynamic = Balanced
+low = Low Power
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
+
+
+### esx_service
+The service name should be in the form of: `ESXi_hostname:<service name`. Eg `esx.example.com:ntpd`
+#### Parameters
+* `running`: true, false
+* `policy`: 'on','off','automatic'
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
+
+### esx_syslog
+#### Parameters
+* `default_rotate`: The maximum number of log files to keep locally on the ESXi host in the configured logDir. Does not affect remote syslog server retention. Defaults to 8
+* `default_size`: The maximum size, in kilobytes, of each local log file before it is rotated. Does not affect remote syslog server retention. Defaults to 1024 KB.
+* `log_dir`: A location on a local or remote datastore and path where logs are saved to. Has the format `[DatastoreName] DirectoryName/Filename`, which maps to `/vmfs/volumes/DatastoreName/DirectoryName/Filename`. The `[DatastoreName]` is case sensitive and if the specified DirectoryName does not exist, it will be created. If the datastore path field is blank, the logs are only placed in their default location. If `/scratch` is defined, the default is `[]/scratch/log`.
+* `log_host`:A remote server where logs are sent using the syslog protocol. If the logHost field is blank, no logs are forwarded. Include the protocol and port, similar to `tcp://hostname:514`
+* `log_dir_unique`: A boolean option which controls whether a host-specific directory is created within the configured logDir. The directory name is the hostname of the ESXi host. A unique directory is useful if the same shared directory is used by multiple ESXi hosts. Defaults to false.
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
+
+### esx_system_resource
+#### Parameters
+This resource allows the configuration of system resources of a host that are viewed und er the 'System Resource Allocation' section of the vSphere client
+* `host`:
+* `system_resource`:
+* `cpu_limit`:
+* `cpu_reservation`:
+* `cpu_expandable_reservation`:
+* `memory_limit`:
+* `memory_reservation`:
+* `memory_expandable_reservation`:
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
+
+### esx_timezone
+#### Parameters
+* `key`: 3 letter time zone. Eg: 'GMT'
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
+
+### esx_datastore
+Manage vCenter esx hosts' datastore.
+#### Parameters
+* `ensure`: present
+* `type`: vmfs, cifs, nfs
+* `lun`: LUN number of storage volume.  Specify only for block storage.
+* `transport`: A resource reference to a transport type declared elsewhere. Eg: `Transport['vcenter']`
+
+### esx_vmknic_type
+Manages ESXi vmknic types - management, vmotion, faultToleranceLogging, or vSphereReplication
```

Comment:
This one munges the ESXi host and nicname parameters from the resource name. It would be nice to call that out here as you did for esx_service.

For more see, https://github.com/vmware/vmware-vcenter/blob/master/lib/puppet/type/esx_vmknic_type.rb#L13-L20
"{ESX Host}:{name of vmknic}"